numba/np/arrayobj.py

```diff
@@ -41,6 +41,28 @@
                                        _choose_concatenation_layout)
 
 
+@lower_builtin(operator.is_, types.Record, types.Record)
```

Comment:
Rather than inserting this lowering right at the top of the file, the natural place for it would be in the `Comparisons` section, next to the implementation of `array_is`.